### DISCUSSION TITLE
Test Class LoggerEngineDataSelector_Tests throwing error

### DISCUSSION THREAD

**Discussion by adityavgpl** [jongpie/NebulaLogger#358] [Q&A] [answered]
Hi All,

I am using managed package v4.8.0 of Nebula, the issue is while running tests in **LoggerEngineDataSelector_Tests** I am getting a failure for the below test case:

`static void it_returns_cached_auth_session()`

this method is throwing an **ERROR** like below:

> System.AssertException: Assertion Failed: Expected: null, Actual: AuthSession:{Id=0AkS000002xIdRpJAM, LoginType=Application, LoginHistoryId=0YaS000003NsaCjTAJ, SessionSecurityLevel=STANDARD, SessionType=UI}

Can you anyone please suggest me how to resolve this, its sort of urgent !!

cc: @jongpie

**Comment by jongpie** [comment, marked as answer]
Hi @adityavgpl - I'll have to figure out why that test is failing, it worked in my scratch orgs & during the packaging process, but the `AuthSession` stuff is always a little problematic to deal with.

Since you're using the managed package, there's not a way for you to resolve this right now - but how is this impacting you? Tests for managed packages shouldn't prevent you from running your own deployments, so I just want to understand the impact of this.

**Reply by adityavgpl** [reply]
I actually got the error while running tests manually. I was running each tests in Nebula package to ensure the my deployment is not failing because of test class failure inside Nebula package.

Deployment did not fail because of the above mentioned test class, so we are good here @jongpie .

**Reply by jongpie** [reply]
@adityavgpl glad to hear it! I was able to fix the test failure in the unlocked package (part of the upcoming release `v4.8.2`), and the managed package will get the same fix next month as part of `v4.9.0` Winter '23 release
